sdk/tables/azure-data-tables/samples/sample_copy_table.py

```diff
@@ -0,0 +1,129 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_copy_table.py
+
+DESCRIPTION:
+    These samples demonstrate how to copy tables between table and blob storage.
+
+USAGE:
+    python sample_copy_table.py
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_STORAGE_CONNECTION_STRING - the connection string to your storage account
+    2) TABLES_STORAGE_ENDPOINT_SUFFIX - the Table service account URL suffix
+    3) TABLES_STORAGE_ACCOUNT_NAME - the name of the storage account
+    4) TABLES_PRIMARY_STORAGE_ACCOUNT_KEY - the storage account access key
+"""
+
+
+from datetime import datetime
+from uuid import uuid4, UUID
+import copy
+import os
+import json
+from azure.storage.blob import BlobServiceClient
+from azure.data.tables import TableServiceClient
+from dotenv import find_dotenv, load_dotenv
+
+
+class CopyTableSamples(object):
+    def __init__(self):
+        load_dotenv(find_dotenv())
+        self.access_key = os.getenv("TABLES_PRIMARY_STORAGE_ACCOUNT_KEY")
+        self.endpoint_suffix = os.getenv("TABLES_STORAGE_ENDPOINT_SUFFIX")
+        self.account_name = os.getenv("TABLES_STORAGE_ACCOUNT_NAME")
+        self.table_connection_string = "DefaultEndpointsProtocol=https;AccountName={};AccountKey={};EndpointSuffix={}".format(
+            self.account_name, self.access_key, self.endpoint_suffix
+        )
+        table_name = "mytable"
+        self.table_service_client = TableServiceClient.from_connection_string(conn_str=self.table_connection_string, table_name=table_name)
+        self.blob_connection_string = os.getenv("AZURE_STORAGE_CONNECTION_STRING")
+        self.blob_service_client = BlobServiceClient.from_connection_string(self.blob_connection_string)
+        self.container_name = table_name
+        self.entity = {
+            "PartitionKey": "color",
+            "text": "Marker",
+            "color": "Purple",
+            "price": 4.99,
+            "last_updated": datetime.today(),
+            "product_id": uuid4(),
+            "inventory_count": 42,
+            "barcode": b"135aefg8oj0ld58"
+        }
+
+    def copy_table_from_table_to_blob(self):
+        self._setup_table()
+        try:
+            self.container_client = self.blob_service_client.create_container(self.container_name)
+            # Download entities from table to memory
+            entities = self.table_client.list_entities()
+            # Upload in-memory table data to a blob that stays in a container
+            for i in range(10):
+                entity = entities[i]
```

Comment:
The entities need to be iterated first - so I think we need `for entity in client.list_entities()`